pkg/appgw/frontend_listeners.go

```diff
@@ -115,10 +114,14 @@ func (c *appGwConfigBuilder) getListenerConfigs(cbCtx *ConfigBuilderContext) map
 	allListeners := make(map[listenerIdentifier]listenerAzConfig)
 	for _, ingress := range cbCtx.IngressList {
 		glog.V(5).Infof("Processing Rules for Ingress: %s/%s", ingress.Namespace, ingress.Name)
+		policy, err := annotations.WAFPolicy(ingress)
+		if len(policy) > 0 {
+			glog.V(5).Infof("Found WAF policy: %s", policy)
+		}
 		azListenerConfigs := c.getListenersFromIngress(ingress, cbCtx.EnvVariables)
 		for listenerID, azConfig := range azListenerConfigs {
-			if cbCtx.EnvVariables.AttachWAFPolicyToListener {
-				attachFirewallPolicy(cbCtx, ingress, &azConfig)
+			if cbCtx.EnvVariables.AttachWAFPolicyToListener || (err == nil && policy != "") {
```

Comment:
should `err == nil` be used here ?

Reply:
> should err == nil be used here ?

The tuple is returned by annotations.WAFPolicy(ingress), we could ignore the err from the return, but I don't see any difference given the intention here?